spec/models/generic_announcement_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe GenericAnnouncement, type: :model do
+  describe '.default_scope' do
+    before do
+      create_list(:system_announcement, 3)
+      create_list(:instance_announcement, 3)
+    end
+
+    it 'orders by descending valid_from within type' do
+      system, instance = GenericAnnouncement.valid.partition { |a| a.is_a? SystemAnnouncement }
+
+      system_dates = system.map(&:valid_from)
+      sorted_dates = system_dates.sort { |a, b| b <=> a }
+      expect(system_dates).to eq(sorted_dates)
+
+      instance_dates = instance.map(&:valid_from)
+      sorted_dates = instance_dates.sort { |a, b| b <=> a }
+      expect(instance_dates).to eq(sorted_dates)
+    end
+
+    it 'orders system announcements before instance announcements' do
+      announcements = GenericAnnouncement.valid
```

Comment:
I would personally use a method like `is_ordered_by` (with a hash param for ascending/descending), but there doesn't seem to be one provided by default. If I had to implement one, would it make sense to extend `Enumerable` with it? Or should I just implement it locally as a sort of helper function?